diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter5/ParameterNameInOuterLambdaRule.kt

```diff
@@ -0,0 +1,46 @@
+package org.cqfn.diktat.ruleset.rules.chapter5
+
+import com.pinterest.ktlint.core.ast.ElementType
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.PARAMETER_NAME_IN_OUTER_LAMBDA
+import org.cqfn.diktat.ruleset.rules.DiktatRule
+import org.cqfn.diktat.ruleset.utils.findAllDescendantsWithSpecificType
+import org.cqfn.diktat.ruleset.utils.findAllNodesWithCondition
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+/**
+ * Rule 5.2.7 check parameter name in outer lambda
+ */
+class ParameterNameInOuterLambdaRule(configRules: List<RulesConfig>) : DiktatRule(
+    "parameter-name-in-outer-lambda",
+    configRules,
+    listOf(PARAMETER_NAME_IN_OUTER_LAMBDA)
+) {
+
+    override fun logic(node: ASTNode) {
+        if (node.elementType == ElementType.LAMBDA_EXPRESSION) {
+            checkLambda(node)
+        }
+    }
+
+    private fun checkLambda(node: ASTNode) {
+        val copyNode = node.clone() as ASTNode
+
+        val lambdaCount = copyNode.findAllNodesWithCondition { it.elementType == ElementType.LAMBDA_EXPRESSION }.size
+        if (lambdaCount > 1) {
+            copyNode.findAllNodesWithCondition { it.elementType == ElementType.LAMBDA_EXPRESSION }
```

Comment:
Yeah, let's extract and rework this logic  - add comments, make sure that we really need some `copyNodes`, which we for some reasons modify

For now it not really clear for me, what we do here
We looking for existence of `it` keyword and looking for some `VALUE_PARAMETER_LIST`, what for?